spring-kafka/src/main/java/org/springframework/kafka/support/MimeTypeJsonDeserializer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.support;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.util.MimeType;
+
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdNodeBasedDeserializer;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+
+/**
+ * {@link MimeType} deserializer.
+ *
+ * @param <T> the type.
+ *
+ * @author Gary Russell
+ * @since 1.3
+ *
+ */
+public class MimeTypeJsonDeserializer<T extends MimeType> extends StdNodeBasedDeserializer<T> {
```

Comment:
Not sure what made you to have this strange generics configuration, but simplified version works well for me:
https://github.com/artembilan/spring-kafka/commit/5511ad52ab530c40d46fbdc5c289ce6b6d798a67

Am I missing anything?
Thanks

Reply:
I had that originally but had some problem - must have been pilot error; feel free to use your version.